client/foundries.go

```diff
@@ -703,6 +716,37 @@ func (a *Api) TargetImageCreate(factory string, targetName string, appShortlist
 	return getResponse(resp, err, "assemble-system-image")
 }
 
+// Return a Compose App for a given Target by a Target ID and an App name
+func (a *Api) TargetComposeApp(factory string, targetName string, app string) (*ComposeAppInspect, error) {
+	url := a.serverUrl + "/ota/factories/" + factory + "/targets/" + targetName + "/compose-apps/" + app + "/"
+	log := logrus.WithFields(logrus.Fields{"url": url})
+	logrus.Debugf("TargetApp with url: %s", url)
+
+	resp, err := a.RawGet(url, nil)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	} else if resp.StatusCode != 200 {
+		log.WithFields(logrus.Fields{"status": resp.StatusCode}).Error("HTTP Error")
+		// Don't return, we can get some debug info from response.
```

Comment:
I don't think you can unmarshall *all* http errors as a ComposeAppInspect below.

Reply:
yes, that's true... but if it fails to read content or unmarshall it - we'll return an error at lines 738 and 744 respectively.

Reply:
Actually, the only few cases when we cannot unmarshall are:
- nginx side error (e.g. 502 when ota-lite is down or smth like that)
- an internal application error
- a 400 or 404 error in case if fioctl does something insane

In all other cases ota-lite will return ComposeAppBundle, just some fields may be empty.

Reply:
but instead of returning an http error, you'll return a marshalling error and the user wouldn't know the cause, right? i.e maybe this should be captured as "orig-err" and return that if marshalling fails?

Reply:
There are several things to it:
- first, I tried to keep the same logic as in other places, the only deviation is to try unmarshal in case of a "sane" HTTP error
- second, a user is able to see all the details with a -v option.

So, I didn't want to complicate the logic with many if statements, though I think that's possible.
Will try to return an original HTTP error in case if unmarshall fails (or combine both into one error).

Reply:
This is what I currently get in case if unmarshal fails after an HTTP error:
```
vkhoroz@osf-ubuntu:~/dev/fioctl$ ./bin/fioctl-linux-amd64 -c ./bin/config.yml targets -f msul-dev01 show compose-app 617 app-07
ERRO[0002] HTTP Error                                    status="502 Bad Gateway" url="https://api.foundries.io/ota/factories/msul-dev01/targets/intel-corei7-64-lmp-617/compose-apps/app-07/"
ERRO[0002] invalid character '\x11' looking for beginning of value  url="https://api.foundries.io/ota/factories/msul-dev01/targets/intel-corei7-64-lmp-617/compose-apps/app-07/"
ERROR:invalid character '\x11' looking for beginning of value
```
I think it's good enough, isn't it ?

Reply:
Changed this to be more informative:
```
ERRO[0002] HTTP Error: 502 Bad Gateway                   status=502 url="https://api.foundries.io/ota/factories/msul-dev01/targets/raspberrypi4-64-lmp-617/compose-apps/app-07/"
ERRO[0002] Parse Error: invalid character 'g' in literal null (expecting 'u')  url="https://api.foundries.io/ota/factories/msul-dev01/targets/raspberrypi4-64-lmp-617/compose-apps/app-07/"
ERROR: Parse Error: invalid character 'g' in literal null (expecting 'u')
 for 'https://api.foundries.io/ota/factories/msul-dev01/targets/raspberrypi4-64-lmp-617/compose-apps/app-07/' HTTP response: 502 Bad Gateway
=nginx bad gateway html response
```